fitbenchmarking/fitbenchmarking
# Added a more interactive (Dash) version of performance profiles

## Description of work

Fixes #1194 .

Main changes:
- Added Dash app within fitbenchmarking, to be run after the results are produced. Unless the user stops the app, the app keeps running after fitbenchmarking has finished.
- Added Dash plots to the accuracy and runtime tables - these are only available if the app is running
  - The user can switch between the new Dash plots and the old plotly plots (always available, no matter whether Dash is running), by clicking a button.
  - Dash plots are more interactive than plotly plots - however, in this PR the only added interactivity available in the Dash plots is that the user is able to switch between linear and log x-axis scale.
- Modified docs to inform the user of the changes
- Added option to specify whether to run the app or not (default is yes), which can be done through the command line or options file
- Added option to specify which port to use to run dash.





## Testing Instructions



First run with "fitbenchmarking" only, to check Dash (Dash will run by default). Can check:
1. That you have two buttons (one for Online and one for Offline graphs), and they work
2. That in the Dash plot, one can change the x axis
3. Try stopping Dash to see what happens (changing log axis stops working, and after changing to offline plot, one cannot obtain the online one any more)

Could then check the new options through the command line:
1. --run_dash and --dont_run_dash
2. --port for specifying the Dash port

Then one could check that all works when passing more than one problem set. For example using:
fitbenchmarking   -o examples/options_template.ini -p examples/benchmark_problems/NIST/low_difficulty  examples/benchmark_problems/NIST/average_difficulty .

Could also check that the option --load-checkpoint still works too, as that code was modified too.

## Checklist
